distributed/worker.py

```diff
@@ -1441,7 +1443,7 @@ def add_task(
                 self.tasks[key] = ts = TaskState(
                     key=key, runspec=SerializedTask(function, args, kwargs, task)
                 )
-                ts.state = "waiting"
+                ts.state = "waiting" if not speculative else "speculative"
```

Comment:
Hrm, what if the dependency has just finished running?  Maybe there is a way to trust the local state here more than what the scheduler said.

Reply:
It shouldn't make a difference, when `add_task` finishes we'll have noted that the dependency is in memory and then the task will transition from speculative -> ready